src/deltacode/cli_test_utils.py

```diff
@@ -0,0 +1,210 @@
+import io
+import json
+import os
+
+from commoncode.system import on_linux
+from commoncode.system import on_windows
+from scancode_config import scancode_root_dir
+
+
+def run_scan_plain(options, cwd=None, test_mode=True, expected_rc=0, env=None):
+    """
+    Run a scan as a plain subprocess. Return rc, stdout, stderr.
+    """
+    from commoncode.command import execute2
+
+    options = add_windows_extra_timeout(options)
+
+    if test_mode and '--test-mode' not in options:
+        options.append('--test-mode')
+
+    scmd = b'deltacode' if on_linux else 'deltacode'
+    scan_cmd = os.path.join(scancode_root_dir, scmd)
+    rc, stdout, stderr = execute2(cmd_loc=scan_cmd, args=options, cwd=cwd, env=env)
+
+    if rc != expected_rc:
+        opts = get_opts(options)
+        error = '''
+Failure to run: deltacode %(opts)s
+stdout:
+%(stdout)s
+
+stderr:
+%(stderr)s
+''' % locals()
+        assert rc == expected_rc, error
+
+    return rc, stdout, stderr
+
+
+def run_scan_click(options, monkeypatch=None, test_mode=True, expected_rc=0, env=None):
+    """
+    Run a scan as a Click-controlled subprocess
+    If monkeypatch is provided, a tty with a size (80, 43) is mocked.
+    Return a click.testing.Result object.
+    """
+    import click
+    from click.testing import CliRunner
+    from deltacode import cli
+
+    options = add_windows_extra_timeout(options)
+
+    if monkeypatch:
+        monkeypatch.setattr(click._termui_impl, 'isatty', lambda _: True)
+        monkeypatch.setattr(click , 'get_terminal_size', lambda : (80, 43,))
+    runner = CliRunner()
+
+    result = runner.invoke(cli.cli, options, catch_exceptions=False, env=env)
+
+    output = result.output
+    if result.exit_code != expected_rc:
+        opts = get_opts(options)
+        error = '''
+Failure to run: deltacode %(opts)s
+output:
+%(output)s
+''' % locals()
+        assert result.exit_code == expected_rc, error
+    return result
+
+
+def get_opts(options):
+    try:
+        return ' '.join(options)
+    except:
+        try:
+            return b' '.join(options)
+        except:
+            return b' '.join(map(repr, options))
+
+
+WINDOWS_CI_TIMEOUT = '222.2'
+
+
+def add_windows_extra_timeout(options, timeout=WINDOWS_CI_TIMEOUT):
+    """
+    Add a timeout to an options list if on Windows.
+    """
+    if on_windows and '--timeout' not in options:
+        # somehow the Appevyor windows CI is now much slower and timeouts at 120 secs
+        options += ['--timeout', timeout]
+    return options
+
+
+def remove_windows_extra_timeout(scancode_options, timeout=WINDOWS_CI_TIMEOUT):
+    """
+    Strip a test timeout from a pretty deltacode_options mapping if on Windows.
+    """
+    if on_windows:
+        if scancode_options and scancode_options.get('--timeout') == timeout:
+            del scancode_options['--timeout']
+
+
+def check_json_scan(expected_file, result_file, regen=False, remove_file_date=False, ignore_headers=False):
```

Comment:
You really only need this function to be able to compare two JSON files for our tests.